test/e2e/metrics_test.go

```diff
@@ -89,134 +129,265 @@ func NewMetricsTestConfig(t *testing.T, client, selector, clusterRole, clusterBi
-		"--restart=Never",
-		"--overrides", `{
-			"spec": {
-				"terminationGradePeriodSeconds": 0,
-				"containers": [{
-					"name": "curl",
-					"image": "curlimages/curl",
-					"command": ["sh", "-c", "sleep 3600"],
-					"securityContext": {
-						"allowPrivilegeEscalation": false,
-						"capabilities": {"drop": ["ALL"]},
-						"runAsNonRoot": true,
-						"runAsUser": 1000,
-						"seccompProfile": {"type": "RuntimeDefault"}
-					}
-				}],
-				"serviceAccountName": "`+c.serviceAccount+`"
-			}
-		}`)
-	output, err := cmd.CombinedOutput()
-	require.NoError(c.t, err, "Error creating curl pod: %s", string(output))
+
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      c.curlPodName,
+			Namespace: c.namespace,
+		},
+		Spec: corev1.PodSpec{
+			ServiceAccountName:            c.serviceAccount,
+			TerminationGracePeriodSeconds: int64Ptr(0),
+			Containers: []corev1.Container{
+				{
+					Name:    "curl",
+					Image:   "curlimages/curl",
+					Command: []string{"sh", "-c", "sleep 3600"},
+					SecurityContext: &corev1.SecurityContext{
+						AllowPrivilegeEscalation: boolPtr(false),
+						Capabilities: &corev1.Capabilities{
+							Drop: []corev1.Capability{"ALL"},
+						},
+						RunAsNonRoot: boolPtr(true),
+						RunAsUser:    int64Ptr(1000),
+						SeccompProfile: &corev1.SeccompProfile{
+							Type: corev1.SeccompProfileTypeRuntimeDefault,
+						},
+					},
+				},
+			},
+			RestartPolicy: corev1.RestartPolicyNever,
+		},
+	}
+
+	_, err := c.kubeClient.CoreV1().Pods(c.namespace).Create(ctx, pod, metav1.CreateOptions{})
+	require.NoError(c.t, err, "Error creating curl pod")
 }
 
-// validate verifies if is possible to access the metrics
-func (c *MetricsTestConfig) validate(token string) {
+// waitForPodReady polls until the Pod is in Ready condition
+func (c *MetricsTestConfig) waitForPodReady(ctx context.Context) {
 	c.t.Log("Waiting for the curl pod to be ready")
-	waitCmd := exec.Command(c.client, "wait", "--for=condition=Ready", "pod", c.curlPodName, "-n", c.namespace, "--timeout=60s")
-	waitOutput, waitErr := waitCmd.CombinedOutput()
-	require.NoError(c.t, waitErr, "Error waiting for curl pod to be ready: %s", string(waitOutput))
+	err := wait.PollUntilContextTimeout(ctx, 2*time.Second, 60*time.Second, false, func(ctx context.Context) (bool, error) {
+		pod, err := c.kubeClient.CoreV1().Pods(c.namespace).Get(ctx, c.curlPodName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		for _, cond := range pod.Status.Conditions {
+			if cond.Type == corev1.PodReady && cond.Status == corev1.ConditionTrue {
+				return true, nil
+			}
+		}
+		return false, nil
+	})
+	if err != nil {
+		// If the context timed out, the test should fail with a more direct message
+		if errors.Is(err, context.DeadlineExceeded) {
+			c.t.Fatal("Timed out waiting for the curl pod to become Ready")
+		}
+		require.NoError(c.t, err, "Error waiting for curl pod to become Ready")
+	}
```

Comment:
super nit: you could get rid of one nesting level, by doing:
```go
if errors.Is(err, context.DeadlineExceeded) {
	c.t.Fatal("Timed out waiting for the curl pod to become Ready")
}
if err != nil {
    ....
}
```

as errors.Is() won't do anything if the error is nil